src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts

```diff
@@ -565,6 +566,66 @@ export const selectEarnUserStakingOpportunityByUserStakingId = createDeepEqualOu
   },
 )
 
+// A staking opportunity by opportunityId parsed as an EarnOpportunityType
+// TODO: testme
+export const selectEarnUserStakingOpportunityByOpportunityId = createDeepEqualOutputSelector(
```

Comment:
@NeOMakinG what's the rationale for this guy vs. `selectEarnUserStakingOpportunityByUserStakingId` ? Input selector params look off either, we use `userStakingId` and `OpportunityId` (a `StakingId` here) so we should be able to use the existing one?

i.e `userStakingId` already contains the `OpportunityId` (`StakingId`), which can be deserialized with `deserializeUserStakingId`, and the only difference in those two selectors seems to be `    const stakingOpportunity = stakingOpportunitiesById[opportunityId]` which I suppose was the intent here (lookup by OpportunityId)?

Reply:
See https://github.com/shapeshift/web/pull/7934#discussion_r1803826127

Reply:
Done in [527f86c](https://github.com/shapeshift/web/pull/7934/commits/527f86ce55071bc81f46808931a053d3e5ee04e3)